back/engines/commercial/id_clave_unica/spec/requests/clave_unica_verification_spec.rb

```diff
@@ -141,12 +141,57 @@ def expect_to_create_verified_user(user)
       password_digest: nil
     })
 
-    expect(response).to redirect_to('/en/complete-signup?pathname=%2Fwhatever-page')
+    expect(response).to redirect_to('/en/complete-signup?pathname=%2Fsome-page')
   end
 
-  context 'when phone registration enabled' do
+  context 'when verification is already taken by new user' do
+    before do
+      get '/auth/clave_unica'
+      follow_redirect!
+    end
+
+    let!(:new_user) do
+      User.order(created_at: :asc).last.tap do |user|
+        expect(user).to have_attributes({ email: nil })
+        expect_to_create_verified_user(user)
+      end
+    end
+
+    context 'when verified registration is completed' do
+      before { new_user.update!(email: Faker::Internet.email) }
+
+      it 'does not verify another user and does not delete previously verified new user' do
+        get "/auth/clave_unica?token=#{@token}&pathname=/some-page"
+        follow_redirect!
+
+        expect(response).to redirect_to('/some-page?verification_error=true&error=taken')
+        expect(@user.reload).to have_attributes({
+          verified: false,
```

Comment:
Slightly confused as to why verified would be false if this is when a verified registration is completed?

Reply:
Good point. Indeed, the spec turned out to be too complicated and confusing. I've tried to fix it a bit with better description https://github.com/CitizenLabDotCo/citizenlab/pull/5279/commits/fbc9295e6e7dcdff3c8990419c7ac6edadb992b0

So, the verified registration is completed by `new_user`, not by `@user` (an old user).

Reply:
Ah OK, that does indeed make it clearer